packages/session-replay-browser/src/hooks/beacon.ts

```diff
@@ -0,0 +1,67 @@
+import { getGlobalScope } from '@amplitude/analytics-client-common';
+import { SessionReplayJoinedConfig } from '../config/types';
+import { SessionReplayDestinationSessionMetadata } from '../typings/session-replay';
+import { getServerUrl } from '../helpers';
+
+type BeaconSendFn<T> = (pageUrl: string, payload: T, contentType: 'application/json') => boolean;
+
+/**
+ * For very small payloads it's preferable to use the [Beacon API](https://developer.mozilla.org/en-US/docs/Web/API/Beacon_API).
+ * While it doesn't provide 100% guarantees on sends, it greatly helps with overall reliability and page load performance. As
+ * the Beacon API has a potential to fail due to size constraints we want to fall back to XHR if need be. This is mostly to
+ * be used with 'pagehide' or 'beforeunload' events.
+ */
+export class BeaconTransport<T> {
+  private sendBeacon: BeaconSendFn<T>;
+  private sendXhr: BeaconSendFn<T>;
+  private readonly basePageUrl: string;
+  private readonly context: Omit<SessionReplayDestinationSessionMetadata, 'deviceId'>;
+  private static readonly contentType = 'application/json';
+
+  constructor(context: Omit<SessionReplayDestinationSessionMetadata, 'deviceId'>, config: SessionReplayJoinedConfig) {
+    const globalScope = getGlobalScope();
+    if (globalScope && globalScope.navigator && typeof globalScope.navigator.sendBeacon === 'function') {
+      this.sendBeacon = (pageUrl, payload, _) => {
+        try {
+          // something seems to be broken if we try to send the content type header in chrome
```

Comment:
What do you mean by this comment?

Reply:
For some reason in testing it always sent OPTIONS instead of POST to my test server.

Reply:
Yep:

From: https://developer.mozilla.org/en-US/docs/Web/API/Beacon_API
> Starting in Chrome 59, this method cannot send a Blob whose type is not CORS safelisted. This is a temporary change until a mitigation can be found for the security issues that this creates. For more information, see [bug 720283](https://crbug.com/720283).